### ISSUE TITLE
Waking up after Sleep

### ISSUE THREAD

**Issue by RedEars** [SchmockLord/Hackintosh-Intel-i9-10900k-Gigabyte-Z490-Vision-D#30]
So apparently not everything seems to work on my Hackintosh Build. When I set the mac into sleepmode, it wakes up instantly after fully sleeping (fans off). I followed your guide and have the same hardware. Any idea or method to find out what the reason of this could be?

**Comment by SchmockLord** [comment]
Yes. Go to Settings - Energy saver and remove "Wake for network access" and "Enable Power Nap". Disable the last flag in advanced bluetooth settings "Wake from Bluetooth Devices". Add darkwake=0 or 8 to your bootargs.

**Comment by jamesfawcett** [comment]
@RedEars just wondering if this solved the issue for you? I still have this issue on mine, I've changed the Bluetooth and sleep settings. My log shows `AppleDeviceManagementHIDEventService::processWakeReason Wake reason: Host (0x01) `

I was wondering what message you have in your console when yours instantly reboots?

**Comment by addisonduda** [comment]
> @RedEars just wondering if this solved the issue for you? I still have this issue on mine, I've changed the Bluetooth and sleep settings. My log shows `AppleDeviceManagementHIDEventService::processWakeReason Wake reason: Host (0x01) `
> 
> 
> 
> I was wondering what message you have in your console when yours instantly reboots?
> 
> 
> 
> Just to add to this - When put to sleep, the screen goes dark and stays dark, but the fans and cpu only turn off for a couple of seconds.
> 
> 

James I have the same issue after changing the bluetooth and energy settings. Put to sleep, monitor turns of and fans/cpu just for a second or two, and then it will turn back on. Sometimes the monitor will turn on as well. It will continue to cycle like this every couple minutes unless I interact with the computer. I am losing bluetooth connections to my mouse and keyboard, so those are definitely not waking up the computer.

**Comment by jamesfawcett** [comment]
I've done some more troubleshooting with this today, but still not solved. It's not my USB config, I've recreated the port configurations and disabled HS14 in both boot-args and removed from the SSDT's. 

I found a really helpful command `pmset -g assertions` it shows which processed are preventing sleep. 

I am still getting the error ` AppleDeviceManagementHIDEventService::processWakeReason Wake reason: Host (0x01)`

When I run the pmset command, I see: 

`Idle sleep preventers: IODisplayWrangler`

I'm going to do some more research, it would be interesting to see what you get from `pmset -g assertions`?

**Comment by addisonduda** [comment]
@RedEars and @jamesfawcett were you able to figure this out for yourself? I finally discovered what the issue was for my build. I have the H100i RGB PRO XT. Last night I came across that other similar Corsair AIOs have caused sleep issues. The internal USB 2.0 is reported to be the issue, because the cooler is continuously trying to communicate with the Corsair Link software (which isn't available in macOS anyway...) and wakes the system up. The internal USB is only used to change settings for the cooler and doesn't have an effect operation once you have chosen you preferred settings and are happy with them for relatively long term. So... I just unplugged the internal USB, turned off _Wake for Network Access_ and sleeps works great!

**Comment by jamesfawcett** [comment]
Oh sorry yes! I managed to fix it in the end. Both my USB headers on the motherboard ran through one hub, I couldn't disable just one of them because I need the other on the same hub for my bluetooth (Fenvi T919). To resolve, I ran a micro USB cable to the Corsair iCUE USB to a USB port on the back of the machine. Not the prettiest solution, but at least I could map that USB port to Internal in mac, and still have my bluetooth working from the internal header. All resolved thank you https://github.com/SchmockLord/Hackintosh-Intel-i9-10900k-Gigabyte-Z490-Vision-D/issues/61

**Comment by JohnyDeejay** [comment]
> Yes. Go to Settings - Energy saver and remove "Wake for network access" and "Enable Power Nap". Disable the last flag in advanced bluetooth settings "Wake from Bluetooth Devices". Add darkwake=0 or 8 to your bootargs.

Hello. I have similar board Asus prime z490a but the only thing which is not working is sleep. I tried all your modifications but without any luck. I always get a restart after sleep. Is there any other solution to check maybe?

Thank you in advance.
Kind regards,
Johnny

**Comment by jamesfawcett** [comment]
Hi @JohnyDeejay can you search your system.log in the console for "Wake Reason:" and let us know what you find?

**Comment by JohnyDeejay** [comment]
I got this messages in iMac logs but in system log i got none when I searched wake or wake reason etc.

<img width="2559" alt="Screenshot 2020-11-05 at 12 36 03" src="https://user-images.githubusercontent.com/73950676/98236328-7af48280-1f63-11eb-84cf-8ad28ff77323.png">

system log after sleep:
<img width="2553" alt="Screenshot 2020-11-05 at 12 48 18" src="https://user-images.githubusercontent.com/73950676/98237472-38cc4080-1f65-11eb-8339-c18e464453a6.png">

**Comment by danny-arroway** [comment]
> @RedEars and @jamesfawcett were you able to figure this out for yourself? I finally discovered what the issue was for my build. I have the H100i RGB PRO XT. Last night I came across that other similar Corsair AIOs have caused sleep issues. The internal USB 2.0 is reported to be the issue, because the cooler is continuously trying to communicate with the Corsair Link software (which isn't available in macOS anyway...) and wakes the system up. The internal USB is only used to change settings for the cooler and doesn't have an effect operation once you have chosen you preferred settings and are happy with them for relatively long term. So... I just unplugged the internal USB, turned off _Wake for Network Access_ and sleeps works great!

Thank you so much for such a wonderful suggestion, really appreciate this!
I have a Corsair 280X Crystal Series 280X RGB case and it too required a USB 2.0 header to connect to the motherboard. 
It was detected as internal port 13 (type 255 when I did my USB Port Mapping), and you're right, it was causing all the immediate wake from sleep issues!
I disconnected the USB 2.0 header, but also had to remap my USB ports to exclude the internal port 13, and now everything works beautifully on my 11600K running Sonoma 14.5.
Thank you!
